react/DeviceCoordinates.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useState, useEffect, useCallback } from 'react'
+import GET_ADDRESS_FROM_GEOCOORDINATES from './graphql/queries.graphql'
+import { useAddressContext } from 'vtex.address-context/AddressContext'
+import { ButtonPlain, Spinner, Tooltip, IconLocation } from 'vtex.styleguide'
+import { FormattedMessage } from 'react-intl'
+import { useLazyQuery } from 'react-apollo'
+
+enum State {
+  PROMPT,
+  GRANTED,
+  LOADING,
+  DENIED,
+}
+
+const DeviceCoordinates: StorefrontFunctionComponent<{}> = () => {
+  const { setAddress } = useAddressContext()
+  const [state, setState] = useState<State>(State.PROMPT)
+  const [getAddressFromGeocoordinates, { error, loading, data }] = useLazyQuery(
+    GET_ADDRESS_FROM_GEOCOORDINATES
+  )
+
+  if (data) {
+    setAddress(data.reverseGeocode)
+  }
+
+  if (error) {
+    console.warn(`error ${error.message}`)
+  }
+
+  const onGetCurrentPositionSuccess = useCallback(
+    ({ coords }: Position) => {
+      console.log('Success!')
+      getAddressFromGeocoordinates({
+        variables: {
+          lat: coords.latitude.toString(),
+          lng: coords.longitude.toString(),
+          apiKey: 'PUT API KEY HERE',
+        },
+      })
+
+      setState(State.GRANTED)
+    },
+    [getAddressFromGeocoordinates]
+  )
+
+  const onGetCurrentPositionError = useCallback((err: PositionError) => {
+    setState(State.DENIED)
+    console.warn(`ERROR(${err.code}): ${err.message}`)
+  }, [])
+
+  const onButtonClick = useCallback(() => {
+    setState(State.LOADING)
+    navigator.geolocation.getCurrentPosition(
+      onGetCurrentPositionSuccess,
+      onGetCurrentPositionError,
+      {
+        enableHighAccuracy: true,
+      }
+    )
+  }, [onGetCurrentPositionSuccess, onGetCurrentPositionError])
+
+  useEffect(() => {
+    navigator.permissions
+      .query({ name: 'geolocation' })
+      .then((result: { state: string }) => {
+        if (result.state === 'granted') {
```

Comment:
You could use a `const` instead of the string here.

Reply:
these strings are from the Permissions API, and even the [MDN examples](https://developer.mozilla.org/en-US/docs/Web/API/Permissions#Example) show you to use them like this. would you still want this to be refactored to a constant?

Reply:
It's ok for me!